src/libsystemd-network/network-internal.c

```diff
@@ -272,25 +273,35 @@ int config_parse_hwaddr(const char *unit,
                         void *userdata) {
         struct ether_addr **hwaddr = data;
         struct ether_addr *n;
-        int r;
+        int r, before = 0, after = 0;
+        const char *start = rvalue, *end;
 
         assert(filename);
         assert(lvalue);
         assert(rvalue);
         assert(data);
 
+        while (isspace(*start))
+                start++;
+
         n = new0(struct ether_addr, 1);
         if (!n)
                 return log_oom();
 
-        r = sscanf(rvalue, "%02hhx:%02hhx:%02hhx:%02hhx:%02hhx:%02hhx",
+        r = sscanf(start, "%n%02hhx:%02hhx:%02hhx:%02hhx:%02hhx:%02hhx%n",
+                   &before,
                    &n->ether_addr_octet[0],
                    &n->ether_addr_octet[1],
                    &n->ether_addr_octet[2],
                    &n->ether_addr_octet[3],
                    &n->ether_addr_octet[4],
-                   &n->ether_addr_octet[5]);
-        if (r != 6) {
+                   &n->ether_addr_octet[5],
+                   &after);
+        end = start + after;
+        if (r == 6)
+                while (isspace(*end))
+                        end++;
+        if ((r != 6) || ((after-before) != 17) || *end != '\0') {
```

Comment:
So, this cements that only 00:01:02:03:04:05 is a valid MAC address, but not 0:1:2:3:4:5? I am think we should probably be more relaxing here, in particular as ether_ntoa() actually outputs the addresses without 0 prefix, iirc, and we should probably be able to parse what glibc spits out...